src/ast.c

```diff
@@ -627,20 +620,8 @@ static value_t julia_to_scm_(fl_context_t *fl_ctx, jl_value_t *v)
 {
     if (jl_is_symbol(v))
         return symbol(fl_ctx, jl_symbol_name((jl_sym_t*)v));
-    if (jl_is_gensym(v)) {
-        size_t idx = ((jl_gensym_t*)v)->id;
-        size_t i;
-        arraylist_t *jlgensym_to_flisp = &jl_ast_ctx(fl_ctx)->gensym_to_flisp;
-        for (i = 0; i < jlgensym_to_flisp->len; i+=2) {
-            if ((ssize_t)jlgensym_to_flisp->items[i] == idx)
-                return fl_list2(fl_ctx, jl_ast_ctx(fl_ctx)->jlgensym_sym, fixnum((size_t)jlgensym_to_flisp->items[i+1]));
-        }
-        arraylist_push(jlgensym_to_flisp, (void*)idx);
-        value_t flv = fl_applyn(fl_ctx, 0, symbol_value(symbol(fl_ctx, "make-jlgensym")));
-        assert(iscons(flv) && car_(flv) == jl_ast_ctx(fl_ctx)->jlgensym_sym);
-        arraylist_push(jlgensym_to_flisp, (void*)(size_t)numval(car_(cdr_(flv))));
-        return flv;
-    }
+    if (jl_is_ssavalue(v))
+        return fl_list2(fl_ctx, jl_ast_ctx(fl_ctx)->ssavalue_sym, fixnum((size_t)((jl_ssavalue_t*)v)->id));
```

Comment:
I don't understand the use case for `expand(SSAValue(3))`. Plus, before this change it also gave `GenSym(0)`. But ok, I'm willing to give an error here.

I'm thinking, for example, that SSAValues could become statement numbers in a fully-linear IR. We might even want to do this now. (Then we wouldn't need explicit assignment expressions for ssavalues, and Exprs wouldn't need the `typ` field.) Then they would refer to indexes in the function body they belong to, and would have no meaning in isolation.